src/client.py

```diff
@@ -0,0 +1,362 @@
+import requests
+from src.utilities import *
+
+# Production MERMAID API root URL.
+api_url = 'https://api.datamermaid.org/v1/'
+# Development MERMAID API root URL.
+api_dev_url = 'https://dev-api.datamermaid.org/v1/'
+
+
+class Client:
+    """
+    Client base class used for accessing MERMAID API.
+    """
+    # Class variables.
+    # Non-project resource endpoints.
+    npr_endpoints = ['health', 'managements', 'me', 'profiles', 'projecttags', 'sites', 'summarysites', 'version']
+    # Attributes.
+    attrs_endpoints = ['benthicattributes', 'fishfamilies', 'fishgenera', 'fishgroupings', 'fishsizes', 'fishspecies']
+    # Project resources.
+    proj_resources = ['collectrecords', 'managements', 'observers', 'project_profiles', 'sites']
+    # Project observations.
+    proj_obs = ['obsbenthiclits', 'obsbenthicpits','obscoloniesbleached', 'obshabitatcomplexities',
+                'obstransectbeltfishs', 'obsquadratbenthicpercent']
+    # Project sample units, methods and events.
+    samples = ['benthiclittransectmethods', 'benthicpittransectmethods', 'benthictransects', 'beltfishtransectmethods',
+               'bleachingquadratcollectionmethods', 'fishbelttransects', 'habitatcomplexitytransectmethods',
+               'quadratcollections', 'sampleevents', 'sampleunitmethods']
+
+    def __init__(self, token=None, url=api_dev_url):
```

Comment:
You might want to add on `*args, **kwargs` as the last two arguments, to provide for future properties.